### PR TITLE
execute command completion for Record and Playfile command on ChannelExecuteComplete event

### PR DESCRIPTION
**Pull request by vincent-seville**

### CODE REVIEW

File: src/main/scala/esl/FSConnection.scala [hunk 1]
```diff
@@ -902,6 +902,72 @@ abstract class FSConnection extends StrictLogging {
             })
             .mkString("\n")}""".stripMargin
         )
+      case (Some(appId), Some(commandToQueue), _, Some(EventNames.ChannelExecuteComplete)) =>
```
Comment (geekbytes-io) [inline src/main/scala/esl/FSConnection.scala]: case (
            Some(appId),
            Some(
              CommandToQueue(command: Record, executeEvent, executeComplete)
            ),
            _,
            Some(EventNames.ChannelExecuteComplete)
          ) if appId == command.eventUuid =>

Comment (geekbytes-io) [reply src/main/scala/esl/FSConnection.scala]: use this format

Comment (geekbytes-io) [reply src/main/scala/esl/FSConnection.scala]: @vincent-seville

**Review by OlenaBiloshapka — APPROVED** [review]

File: src/main/scala/esl/FSConnection.scala [hunk 2]
```diff
@@ -837,6 +837,36 @@ abstract class FSConnection extends StrictLogging {
     * @return EventMessage
     */
   private def handleFSEventMessage(eventMessage: EventMessage): EventMessage = {
+    def completeAndRemoveFromMap(command: FSExecuteApp, executeComplete: Promise[EventMessage]): Unit = {
```
Comment (geekbytes-io) [inline src/main/scala/esl/FSConnection.scala]: def completeAndRemoveFromMap(command: FSExecuteApp, executeComplete: Promise[EventMessage], canRemove:Boolean): Unit = {
      executeComplete.complete(Success(eventMessage))
      if (canRemove) {
        eventMap.remove(command.eventUuid)
        adapter.info(
          logMarker,
          s"""handleFSEventMessage for app id ${command.eventUuid} Removing entry
             |>> TYPE
             |EVENT ${eventMessage.eventName.getOrElse("NA")}
             |>> HEADERS
             |${
            eventMessage.headers
              .map(h => h._1 + " : " + h._2)
              .mkString(space, "\n" + space, "")
          }
             |>> BODY
             |${eventMessage.body}
             |>> MAP is below
             |${
            eventMap.map({ case (key, value) =>
                s"""appId: $key
                   |command
                   |$value""".stripMargin
              })
              .mkString("\n")
          }""".stripMargin
        )
      }
    }

File: src/main/scala/esl/FSConnection.scala [hunk 3]
```diff
@@ -902,6 +932,20 @@ abstract class FSConnection extends StrictLogging {
             })
             .mkString("\n")}""".stripMargin
         )
+      case (
```
Comment (geekbytes-io) [inline src/main/scala/esl/FSConnection.scala:939]: case (
            Some(appId),
            Some(CommandToQueue(command: Record, execute, executeComplete)),
            _,
            Some(EventNames.ChannelExecuteComplete)
          ) if appId == command.eventUuid =>
        completeAndRemoveFromMap(command, executeComplete, execute.isCompleted)
      case (
            Some(appId),
            Some(CommandToQueue(command: PlayFile, execute, executeComplete)),
            _,
            Some(EventNames.ChannelExecuteComplete)
          ) if appId == command.eventUuid =>
        completeAndRemoveFromMap(command, executeComplete, execute.isCompleted)

**Review by geekbytes-io — CHANGES_REQUESTED** [review]

**Review by geekbytes-io — APPROVED** [review]

**Pull request merged**